builtin/files/stack-templates/node-pool.json.tmpl

```diff
@@ -113,19 +113,52 @@
 {{end}}
 {{define "AutoScaling"}}
     "{{.LogicalName}}": {
+      "DependsOn": "{{.LaunchTemplateLogicalName}}",
       "Properties": {
         "HealthCheckGracePeriod": 600,
         "HealthCheckType": "EC2",
-        "LaunchConfigurationName": {
-          "Ref": "{{.LaunchConfigurationLogicalName}}"
-        },
         "MaxSize": "{{.MaxCount}}",
         "MetricsCollection": [
           {
             "Granularity": "1Minute"
           }
         ],
         "MinSize": "{{.MinCount}}",
+        {{if .AutoScalingGroup.MixedInstances.Enabled }}
+        "MixedInstancesPolicy": {
+          "InstancesDistribution" : {
+            {{if .AutoScalingGroup.MixedInstances.OnDemandAllocationStrategy}}
+            "OnDemandAllocationStrategy" : "{{.AutoScalingGroup.MixedInstances.OnDemandAllocationStrategy}}",
+            {{end}}
+            "OnDemandBaseCapacity" : {{.AutoScalingGroup.MixedInstances.OnDemandBaseCapacity}},
+            "OnDemandPercentageAboveBaseCapacity" : {{.AutoScalingGroup.MixedInstances.OnDemandPercentageAboveBaseCapacity}},
+            {{if .AutoScalingGroup.MixedInstances.SpotAllocationStrategy}}
+            "SpotAllocationStrategy" : "{{.AutoScalingGroup.MixedInstances.SpotAllocationStrategy}}",
+            {{end}}
+            "SpotInstancePools" : {{.AutoScalingGroup.MixedInstances.SpotInstancePools}},
+            "SpotMaxPrice" : "{{.AutoScalingGroup.MixedInstances.SpotMaxPrice}}"
```

Comment:
SpotMaxPrice should be possible to leave out. If you are excluding SpotMaxPrice in the policy AWS will automatically set the on demand price as max price. This is especially important when having an override list of mixed instance types and sizes.

Reply:
I suggest that you add
            {{if .AutoScalingGroup.MixedInstances.SpotMaxPrice}}
            "SpotMaxPrice" : "{{.AutoScalingGroup.MixedInstances.SpotMaxPrice}}"
            {{end}}